data-prepper-plugins/otel-trace-raw-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepperTest.java

```diff
@@ -113,13 +111,98 @@ public void testEmptySpans() {
+        final String sampleRequest2 = "{\"resourceSpans\":[{\"resource\":{\"attributes\":[{\"key\":\"service.name\",\"value\":{\"stringValue\":\"analytics-service\"}},{\"key\":\"telemetry.sdk.language\",\"value\":{\"stringValue\":\"java\"}},{\"key\":\"telemetry.sdk.name\",\"value\":{\"stringValue\":\"opentelemetry\"}},{\"key\":\"telemetry.sdk.version\",\"value\":{\"stringValue\":\"0.8.0-SNAPSHOT\"}}],\"droppedAttributesCount\":0},\"instrumentationLibrarySpans\":[{\"instrumentationLibrary\":{\"name\":\"io.opentelemetry.auto.spring-webmvc-3.1\",\"version\":\"\"},\"spans\":[{\"traceId\":\"bQ/2NNEmtuwsGAOR5ntCNw==\",\"spanId\":\"WsV2GFh9pxw=\",\"traceState\":\"\",\"parentSpanId\":\"mnO/qUT5ye4=\",\"name\":\"LoggingController.save\",\"kind\":\"SPAN_KIND_INTERNAL\",\"startTimeUnixNano\":\"1597902046041803300\",\"endTimeUnixNano\":\"1597902046088892200\",\"attributes\":[],\"droppedAttributesCount\":0,\"events\":[],\"droppedEventsCount\":0,\"links\":[],\"droppedLinksCount\":0,\"status\":{\"code\":\"STATUS_CODE_OK\",\"message\":\"\"}}]},{\"instrumentationLibrary\":{\"name\":\"io.opentelemetry.auto.apache-httpasyncclient-4.0\",\"version\":\"\"},\"spans\":[{\"traceId\":\"bQ/2NNEmtuwsGAOR5ntCNw==\",\"spanId\":\"na9fwApQsYs=\",\"traceState\":\"\",\"parentSpanId\":\"WsV2GFh9pxw=\",\"name\":\"HTTP PUT\",\"kind\":\"SPAN_KIND_CLIENT\",\"startTimeUnixNano\":\"1597902046052809200\",\"endTimeUnixNano\":\"1597902046084822500\",\"attributes\":[{\"key\":\"http.status_code\",\"value\":{\"intValue\":\"200\"}},{\"key\":\"http.url\",\"value\":{\"stringValue\":\"/logs/_doc/service_1?timeout\\u003d1m\"}},{\"key\":\"http.method\",\"value\":{\"stringValue\":\"PUT\"}}],\"droppedAttributesCount\":0,\"events\":[],\"droppedEventsCount\":0,\"links\":[],\"droppedLinksCount\":0,\"status\":{\"code\":\"STATUS_CODE_OK\",\"message\":\"\"}}]}]}]}";
+        final ExportTraceServiceRequest exportTraceServiceRequest1 = buildExportTraceServiceRequestFromJson(sampleRequest1);
+        final ExportTraceServiceRequest exportTraceServiceRequest2 = buildExportTraceServiceRequestFromJson(sampleRequest2);
+        List<Record<String>> processedRecords = new ArrayList<>();
+        List<Future<Collection<Record<String>>>> futures = new ArrayList<>();
+        futures.addAll(submitExportTraceServiceRequests(Collections.singletonList(exportTraceServiceRequest1)));
+        futures.addAll(submitExportTraceServiceRequests(Collections.singletonList(exportTraceServiceRequest2)));
+        for (Future<Collection<Record<String>>> future : futures) {
+            processedRecords.addAll(future.get());
+        }
+        await().atMost(2 * TEST_GC_INTERVAL, TimeUnit.SECONDS).untilAsserted(() -> {
+            List<Future<Collection<Record<String>>>> futureList = submitExportTraceServiceRequests(Collections.emptyList());
+            for (Future<Collection<Record<String>>> future : futureList) {
+                processedRecords.addAll(future.get());
+            }
```

Comment:
What's this emptyList processing for, and why is it in the away block?

Reply:
@wrijeff This is somewhat mimicking the context of [ProcessWorker](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/a6b84f22ada1615e5102f1a2c9c7793d5da2acd5/data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/ProcessWorker.java#L43). ProcessWorker keeps on polling the blockingbuffer that returns empty list if no records are present and then send it to prepper. 

Since there is a delay in ingest and export, I need to put it in the await block so that it will keep on reTrying `doExecute` until the returned futures from `doExecute` will include the processed span sets we expect.